client/build_test.go

```diff
@@ -1084,79 +1084,139 @@ func testClientGatewayExecError(t *testing.T, sb integration.Sandbox) {
 	require.NoError(t, err)
 	defer c.Close()
 
-	id := identity.NewID()
-	st := llb.Image("busybox:latest").Run(
-		llb.Dir("/src"),
-		llb.AddMount("/src", llb.Scratch()),
-		llb.Shlexf("sh -c \"echo %s > output && fail\"", id),
-	).Root()
+	b := func(ctx context.Context, c client.Client) (*client.Result, error) {
+		id := identity.NewID()
+		tests := []struct {
+			Name      string
+			State     llb.State
+			NumMounts int
+			Paths     []string
+		}{{
+			"only rootfs",
+			llb.Image("busybox:latest").Run(
+				llb.Shlexf(`sh -c "echo %s > /data && fail"`, id),
+			).Root(),
+			1, []string{"/data"},
+		}, {
+			"rootfs and readwrite scratch mount",
+			llb.Image("busybox:latest").Run(
+				llb.Shlexf(`sh -c "echo %s > /data && echo %s > /rw/data && fail"`, id, id),
+				llb.AddMount("/rw", llb.Scratch()),
+			).Root(),
+			2, []string{"/data", "/rw/data"},
+		}, {
+			"rootfs and readwrite mount",
+			llb.Image("busybox:latest").Run(
+				llb.Shlexf(`sh -c "echo %s > /data && echo %s > /rw/data && fail"`, id, id),
+				llb.AddMount("/rw", llb.Scratch().File(llb.Mkfile("foo", 0700, []byte(id)))),
```

Comment:
If you would set `llb.ForceNoOutput()` in here I think this test would not work.

Reply:
Confirmed it doesn't work, thanks. The `MountID[1]` becomes scratch when setting `llb.ForceNoOutput`.